### PR TITLE
cellSave fix plus bugfixes

### PR DESCRIPTION
**Pull request by Inviuz**
allows allocation of last byte in memory block
prevents rpcs3 from crashing when closing non existent socket

### CODE REVIEW

File: rpcs3/Emu/Memory/vm.cpp [hunk 1]
```diff
@@ -621,7 +621,7 @@ namespace vm
 		size = ::align(size, 4096);
 
 		// return if addr or size is invalid
-		if (!size || size > this->size || addr < this->addr || addr + size - 1 >= this->addr + this->size - 1)
+		if (!size || size > this->size || addr < this->addr || addr + size - 1 >= this->addr + this->size)
```
Comment (Nekotekina) [inline rpcs3/Emu/Memory/vm.cpp]: This may overflow to zero

File: rpcs3/Emu/Memory/vm.cpp [hunk 2]
```diff
@@ -621,7 +621,7 @@ namespace vm
 		size = ::align(size, 4096);
 
 		// return if addr or size is invalid
-		if (!size || size > this->size || addr < this->addr || addr + size - 1 >= this->addr + this->size - 1)
+		if (!size || size > this->size || addr < this->addr || addr + size - 1 > this->addr + this->size -1)
```
Comment (ExE-Boss) [inline rpcs3/Emu/Memory/vm.cpp]: The last `this->size -1` should be `this->size - 1`.

**Comment by Zangetsu38** [comment]
you can fix my crash caused with sys net in TLOU ?
Caused by "shutdown" instruction"
capture d ecran 2017-04-04 08 31 14

**Comment by paulsapps** [comment]
shutdown is a function not an instruction, and the problem is that the socket is nullptr

**Comment by Zangetsu38** [comment]
Thx crash fixed :)

**Pull request merged**